oauthlib/oauth2/rfc6749/grant_types/base.py

```diff
@@ -15,13 +16,22 @@
 class GrantTypeBase(object):
     error_uri = None
     request_validator = None
+    default_response_mode = 'fragment'
```

Comment:
It looks like this was added to satisfy the default fragment response type in the OpenID spec, but I think this partially conflicts with the defaults defined in the OAuth2 Multiple Response Modes spec.  It may not impact existing OAuth2 flows (at least didn't reflect as an error in the tests) but I'll review to see if this can be less far reaching and scoped more toward OpenID specific requests.